### ISSUE TITLE
Add lookup capability to geo_distance query / Support `lat/lon` as lookup field

### ISSUE THREAD

**Issue by RTodorov** [elastic/elasticsearch#27084]
Since `more_like_this` query does not support geospatial fields, one way of implementing it would be to use a normal query where we could "join" the fields via lookup.

For example, I have a restaurant that makes sushi and I want to find related places nearby. I'd like to be able to do something like this:

```
{
    "query": {
        "bool": {
            "filter": [
                {
                    "terms": {
                        "category": {
                            "id": "sushiplace123",
                            "index": "places",
                            "type": "restaurant",
                            "path": "category"
                        }
                    }
                },
                {
                    "geo_distance": {
                        "address.location": {
                            "id": "sushiplace123",
                            "index": "places",
                            "type": "restaurant",
                            "path": "address.location"
                        },
                        "distance": "1km"
                    }
                }
            ],
            "must": [
                {
                    "match_all": {}
                }
            ]
        }
    }
}
```

This query means that I'm looking for places with the same category as sushiplace123 (sushi) and within a 1km radius.

Currently, the only solution I could find is to query the cluster twice.

Is this something that could be implemented? Also, are you considering adding geospatial capabilities to the ``more_like_this`` query / api?

**Comment by colings86** [comment]
Discussed in FixItFriday and we felt that this is not something that we would like to implement. We support lookups in a few specific places where the data retrieved can be sizeable (such as for indexed shapes in the geo_shape query). In this case the data retrieved is a simple lat/lon point so is not something we would add functionality for.

**Comment by RTodorov** [comment]
Understood for the lookup thing.

What about adding ``geo_distance`` support for the ``more_like_this`` query?

**Comment by colings86** [comment]
We would like to make improvements to the more like this query in the future so feel free to open another issue for this and we can consider it when we come to that work. It will be a while before we start work on improving it but if you open an issue we can get it considered when the time comes.
